saiarcot895/sonic-buildimage
# Update Dell z9332f platform modules for Bullseye and 5.10 kernel.

#### Why I did it
Update Dell z9332f platform modules for Bullseye and 5.10 kernel.
#### How I did it
- Add out of tree pca9548 mux driver to use platform data to mapping i2c bus with front panel port.
- Update function call available in kernel 5.10.
- Update platform-modules-z9332f dependency package to linux-image-5.10.0-8-2-amd64-unsigned.
#### How to verify it
Check the driver is working properly with kernel 5.10 and plug SFP at port 1, 2 QSFP-DD at port 4, 7, 13 and 27 to verify the i2c bus mapping alignment is correct.

Expected:
<html>
<body>


|Module Type | Front Panel Port | I2C Bus Number
|-- | -- | --|
|QSFP-DD |   4 | 13|
|QSFP-DD |   7 | 16
|QSFP-DD | 13 | 22
|QSFP-DD | 27 | 36
|SFP | 1 | 1
|SFP | 2 | 2


</body>
</html>

[z9332f_bullseye_support_test_log.txt](https://github.com/saiarcot895/sonic-buildimage/files/7344083/z9332f_bullseye_support_test_log.txt)
#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog

z9332f bullseye support

#### A picture of a cute animal (not mandatory but encouraged)
